.github/workflows/hermetic_library_generation.yaml

```diff
@@ -24,24 +24,41 @@ jobs:
   library_generation:
     runs-on: ubuntu-latest
     steps:
+    - name: Determine whether the pull request comes from a fork
+      run: |
+        if [[ "${GITHUB_REPOSITORY}" != "${REPO_FULL_NAME}" ]]; then
+          echo "This PR comes from a fork. Skip library generation."
+          echo "SHOULD_RUN=false" >> $GITHUB_ENV
+        else
+          echo "SHOULD_RUN=true" >> $GITHUB_ENV
```

Comment:
Is there a way to exit the workflow early so we don't have to pass this env to all the subsequent steps?

Reply:
Looks like another way of skipping steps is creating a separate job and make the generation job depends on that, just like what we've done in unit/integration tests.

I'll keep it as-is now.